include/cograph_recognition/CographRecognition.hpp

```diff
@@ -0,0 +1,66 @@
+
+
+#pragma once
+
+#include <optional>
+#include <vector>
+
+#include <networkit/base/Algorithm.hpp>
+#include <networkit/components/ConnectedComponents.hpp>
+#include <networkit/graph/Graph.hpp>
+#include <networkit/graph/GraphTools.hpp>
+
+namespace Koala {
+
+
+class CographRecognition : public NetworKit::Algorithm {
+ public:
+    
+    enum class State {
+            UNKNOWN,
+            COMPLEMENT_REDUCIBLE,
+            COND_1,
+            COND_2,
+            COND_3,
+            COND_4,
+            COND_5,
+            COND_6
+         };
+
+    /**
+     * Given an input graph, set up the cograph recognition.
+     *
+     * @param graph The input graph.
+     */
+    explicit CographRecognition(NetworKit::Graph &graph);
+
+    /**
+     * Execute the cograph recognition procedure.
```

Comment:
If you really need to stress that cograph = complement reducible, please write "cograph (i.e. complement reducible)" here

Reply:
now cograph is everywhere